File: Cliente - Servidor.java
//Servidor en Java
import java.io.*;
import java.net.*;

public class Servidor {
    public static void main(String[] args) {
        int puerto = 5408;

        try (ServerSocket servidorSocket = new ServerSocket(puerto)) {
            System.out.println("Servidor esperando conexiones en el puerto " + puerto + "...");
            while (true) {
                try (Socket clienteSocket = servidorSocket.accept();
                     PrintWriter out = new PrintWriter(clienteSocket.getOutputStream(), true);
                     BufferedReader in = new BufferedReader(new InputStreamReader(clienteSocket.getInputStream()))) {

                    String mensajeCliente = in.readLine();
                    if ("requesting_home".equals(mensajeCliente)) {
                        String respuesta = "<html><head><title>Saludo</title></head><body><h1>Bienvenido a la pagina de inicio</h1></body></html>";
                        out.println(respuesta);

                        //Guardamos la respuesta en un archivo llamado home.html
                        try (PrintWriter archivo = new PrintWriter("home.html")) {
                            archivo.println(respuesta);
                            System.out.println("Respuesta guardada en home.html");
                        } catch (IOException e) {
                            e.printStackTrace();
                        }
                    }
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}


//Cliente en Java
import java.io.*;
import java.net.*;

public class Cliente {
    public static void main(String[] args) {
        String host = "localhost";
        int puerto = 5408;

        try (Socket socket = new Socket(host, puerto);
             PrintWriter out = new PrintWriter(socket.getOutputStream(), true);
             BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {

            // Enviar mensaje al servidor
            out.println("requesting_home");

            // Recibir respuesta del servidor
            String respuesta = in.readLine();
            System.out.println("Respuesta del servidor:\n" + respuesta);

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Cliente servidor.py
#Servidor python
import socket

puerto = 5408

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as servidor_socket:
    servidor_socket.bind(('localhost', puerto))
    servidor_socket.listen()

    print(f"Servidor esperando conexiones en el puerto {puerto}...")
    conexion, direccion = servidor_socket.accept()

    with conexion, conexion.makefile('r') as entrada, conexion.makefile('w') as salida:
        mensaje_cliente = entrada.readline().strip()
        if mensaje_cliente == "requesting_home":
            respuesta = """<html><head><title>Saludo</title></head><body><h1>Bienvenido a la pagina de inicio</h1></body></html>"""
            salida.write(respuesta)

            # Guardamos la respuesta en un archivo llamado home.html
            with open("home.html", "w") as archivo:
                archivo.write(respuesta)
            print("Respuesta guardada en home.html")

#Cliente python
import socket

puerto = 5408

with socket.create_connection(('localhost', puerto)) as cliente_socket:
    with cliente_socket.makefile('r') as entrada, cliente_socket.makefile('w') as salida:
        # Enviamos mensaje al servidor
        salida.write("requesting_home\n")
        salida.flush()

        # Recibimos respuesta del servidor
        respuesta = entrada.readline().strip()
        print("Respuesta del servidor:\n" + respuesta)
